airflow/models/trigger.py

```diff
@@ -150,14 +154,32 @@ def submit_event(cls, trigger_id, event, session: Session = NEW_SESSION) -> None
         for task_instance in session.query(TaskInstance).filter(
             TaskInstance.trigger_id == trigger_id, TaskInstance.state == TaskInstanceState.DEFERRED
         ):
-            # Add the event's payload into the kwargs for the task
-            next_kwargs = task_instance.next_kwargs or {}
-            next_kwargs["event"] = event.payload
-            task_instance.next_kwargs = next_kwargs
-            # Remove ourselves as its trigger
-            task_instance.trigger_id = None
-            # Finally, mark it as scheduled so it gets re-queued
-            task_instance.state = TaskInstanceState.SCHEDULED
+            if isinstance(event, BaseTaskEndEvent):
```

Comment:
Can this logic live in the event class instead to eliminate the if-else check? TriggerEvent can have a `handle_submit` (or something) that implements the else block here, and `BaseTaskEndEvent` overrides.

Reply:
sure